utils/phone_numbers.py

```diff
@@ -0,0 +1,41 @@
+from typing import List
+
+
+class PhoneNumberExtractorService:
+    INTERNAL_NUMBER_LENGTH: int = None
+    COUNTRY_CODE: str = None
+
+    def __init__(self, raw_phone: str):
+        self.raw_phone = raw_phone
+
+    def _get_digits(self, origin):
+        digits = [digit for digit in list(origin) if digit in '0123456789']
+        return ''.join(list(digits))
+
+    def get_phone_number_in_e164(self):
+        digits = self._get_digits(self.raw_phone)
+        phones = get_phone_numbers(digits, self.INTERNAL_NUMBER_LENGTH)
+        return list(map(lambda phone: f'{self.COUNTRY_CODE} {phone}', phones))
+
+
+class PolandPhoneNumberExtractorService(PhoneNumberExtractorService):
```

Comment:
Does it mean we should add this for all other spiders too?

P.S. does this sanitization have to be done on the spider side or could be on the data enhancement side? CSVs have a country column so we could probably extrapolate the country code from there (not sure about the length)

Reply:
> Does it mean we should add this for all other spiders too?

it depends
I suggest we have both:
1. in the spider, we can reuse this class when a country with a fixed National Significant Numbers(NSN) length.
1. in the data enhancement, we use the phone number package do the parse with a country code

> CSVs have a country column so we could probably extrapolate the country code from there 

that's a good point, we can have a country code map from it, {'Poland: '+48'} or use a package

> (not sure about the length)

That's the tricky part.

Some country works, for example, French telephone numbers have had 9, then it's easy to reuse PhoneNumberExtractorService. But it has a little different as well:
  - All geographic numbers are dialled in the ten-digit format, even for local calls. 

https://en.wikipedia.org/wiki/Telephone_numbers_in_Poland
https://en.wikipedia.org/wiki/Telephone_numbers_in_France

In another case, Germany has an open telephone numbering plan. There used to be no fixed lengths for either area codes or subscriber telephone numbers, meaning that some subscriber numbers may be as short as two digits.
That will be the hardest one.

Reply:
> in the data enhancement, we use the phone number package do the parse with a country code

Shall we add it in this PR? Or you've already done it in `sanitize_phone`?

> In another case, Germany has an open telephone numbering plan. There used to be no fixed lengths for either area codes or subscriber telephone numbers, meaning that some subscriber numbers may be as short as two digits.

If they don't have a fixed length shall we just combine all digits found into the number (as long as not separated by `,` in case of multiple numbers)?

Reply:
> Shall we add it in this PR? Or you've already done it in sanitize_phone?

yes, I've done.

> If they don't have a fixed length shall we just combine all digits found into the number (as long as not separated by , in case of multiple numbers)?

yes
but in some cases, it's not separated by `,`, for example:
[[tests/test_utils.py](https://github.com/safe-refuge/safeway-data/blob/0258a6ffc0f5412899fd1911efdbefe8970ee13c/tests/test_utils.py#L6-L19)](https://github.com/safe-refuge/safeway-data/blob/0258a6ffc0f5412899fd1911efdbefe8970ee13c/tests/test_utils.py#L6-L19)